config/config.go

```diff
@@ -621,16 +623,19 @@ func (c Cluster) stackConfig(opts StackTemplateOptions, compressUserData bool) (
 		return nil, err
 	}
 
-	compactAssets, err := ReadOrCreateCompactTLSAssets(opts.TLSAssetsDir, KMSConfig{
-		Region:         stackConfig.Config.Region,
-		KMSKeyARN:      c.KMSKeyARN,
-		EncryptService: c.providedEncryptService,
-	})
-	if err != nil {
-		return nil, err
-	}
+	var compactAssets *CompactTLSAssets
 
-	stackConfig.Config.TLSConfig = compactAssets
+	if c.ManageCertificates {
+		compactAssets, err = ReadOrCreateCompactTLSAssets(opts.TLSAssetsDir, KMSConfig{
+			Region:         stackConfig.Config.Region,
+			KMSKeyARN:      c.KMSKeyARN,
+			EncryptService: c.providedEncryptService,
+		})
+		if err != nil {
+			return nil, err
+		}
+		stackConfig.Config.TLSConfig = compactAssets
+	}
```

Comment:
Could you add a test for this?
I believe `TestReadOrCreateCompactTLSAssets` would be an useful reference.